test/functional/test_framework/segwit_addr.py

```diff
@@ -99,9 +99,20 @@ def decode(hrp, addr):
     return (data[0], decoded)
 
 
-def encode(hrp, witver, witprog):
+def encode_segwit_address(hrp, witver, witprog):
     """Encode a segwit address."""
     ret = bech32_encode(hrp, [witver] + convertbits(witprog, 8, 5))
-    if decode(hrp, ret) == (None, None):
+    if decode_segwit_address(hrp, ret) == (None, None):
         return None
     return ret
+
+class TestFrameworkScript(unittest.TestCase):
+    def test_segwit_encode_decode(self):
+        def test_python_bech32(addr):
+            hrp = addr[:4]
+            self.assertEqual(hrp, "bcrt")
+            (witver, witprog) = decode_segwit_address(hrp, addr)
+            self.assertEqual(encode_segwit_address(hrp, witver, witprog), addr)
+
+        test_python_bech32('bcrt1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq3xueyj')
+        test_python_bech32('bcrt1qft5p2uhsdcdc3l2ua4ap5qqfg4pjaqlp250x7us7a8qqhrxrxfsqseac85')
```

Comment:
I'm wondering if it would be better to use the `ADDRESS_BCRT1_UNSPENDABLE` and `ADDRESS_BCRT1_P2WSH_OP_TRUE` constants from `address.py` here.

Reply:
> I'm wondering if it would be better to use the `ADDRESS_BCRT1_UNSPENDABLE` and `ADDRESS_BCRT1_P2WSH_OP_TRUE` constants from `address.py` here.

Good idea!

Reply:
I would do this if the `ADDRESS_BCRT_*` constants were in segwit_addr.py. address.py imports from segwit_addr.py, so adding importing these constants from address.py would be a circular dependency. Other options:

- move the constants to segwit_addr.py. Doesn't seem worth it since a bunch of test files would need to be updated (grep for `ADDRESS_BCRT` to see which test files import these constants from address.py)
- merge address.py and segwit_addr.py into a single file. I think this makes sense as a follow-up (and previously suggested it here: https://github.com/bitcoin/bitcoin/pull/19239#issuecomment-642822740)